x/emissions/proto/emissions/v4/events.proto

```diff
@@ -0,0 +1,84 @@
+syntax = "proto3";
+package emissions.v4;
+
+import "emissions/v3/nonce.proto";
+import "emissions/v3/reputer.proto";
+import "gogoproto/gogo.proto";
+
+option go_package = "github.com/allora-network/allora-chain/x/emissions/types";
+
+// We choose a denormalized schema for events to balance the size and number of
+// events, as well as the complexity of likely downstream write and read
+// patterns. One likely cares about how the topic performs in totality and may
+// want to update topic state after all topic information is in, or in some
+// chronological order. If one normalizs by actor type, then the event volume is
+// likely to be much higher and one would have to wait an undetermined amount of
+// time (how many actors participated
+// == how many events to wait for) to update an topic state-level metric.
+// Furthermore, we know that the size of each of these messages is bounded by
+// the global parameters that bound the max number of actors and topics. If one
+// were to track per-actor-per-type metrics, then one immediately knows that
+// they were included and participated or not in the topic at a particular block
+// height as soon as one event from the topic is recorded.
+
+enum ActorType {
+  ACTOR_TYPE_INFERER_UNSPECIFIED = 0;
+  ACTOR_TYPE_FORECASTER = 1;
+  ACTOR_TYPE_REPUTER = 2;
+}
+
+message EventScoresSet {
+  ActorType actor_type = 1;
+  uint64 topic_id = 2;
+  int64 block_height = 3;
+  repeated string addresses = 4;
+  repeated string scores = 5 [
+    (gogoproto.customtype) = "github.com/allora-network/allora-chain/math.Dec",
+    (gogoproto.nullable) = false
+  ];
+}
+
+message EventRewardsSettled {
+  ActorType actor_type = 1;
+  uint64 topic_id = 2;
+  int64 block_height = 3;
+  repeated string addresses = 4;
+  repeated string rewards = 5 [
+    (gogoproto.customtype) = "github.com/allora-network/allora-chain/math.Dec",
+    (gogoproto.nullable) = false
+  ];
+}
+
+message EventNetworkLossSet {
+  uint64 topic_id = 1;
+  int64 block_height = 2;
+  emissions.v3.ValueBundle value_bundle = 3;
+}
+
+message EventForecastTaskScoreSet {
```

Comment:
I see an EventScoreSet and an EventForecastTaskScoreSet

is EventScoreSet for inferers, should it be called EventInferenceTaskScoreSet then?

Do we fire an event when reputers have their scores changed?


are these scores, or score EMAs?

Reply:
`EventScoresSet` handles all performance scores for `inferes`, `forecasters` and `reputers`. And I created this `EventForecastTaskScoreSet` for forecast task score because this is used at only forecasts.
And all of these are score not EMAs. Do you think we need to emit EMA as well?

Reply:
well EMAs are what are actually used for inclusion on chain, so I would think they're more important than instantaneous scores but I'm also of the opinion that we should just log everything all the time

Reply:
Could actually be a really good idea to emit the EMA score with a boolean that indicates if the actor is part of the active or passive set for that round.

So an event with schema: `actor, topic, epoch nonce, ema score, isActive`

Been getting questions related to being able to tell if one is in the active/passive set.

Reply:
thanks guys
I will add this event